src/OutputEllipseComponent.cpp

```diff
@@ -38,9 +53,152 @@ void OutputEllipseComponent::paint(Graphics &g)
 		{
 			auto line = std::static_pointer_cast<isobus::LineAttributes>(child);
 
-			auto vtColour = parentWorkingSet->get_colour(line->get_background_color());
-			g.setColour(Colour::fromFloatRGBA(vtColour.r, vtColour.g, vtColour.b, 1.0f));
-			g.drawEllipse(0, 0, get_width(), get_height(), line->get_width());
+			auto lineColour = parentWorkingSet->get_colour(line->get_background_color());
+			if (useLineColourForFill)
+			{
+				fillColour = lineColour;
+			}
+
+			float centerX = get_width() / 2.0f;
+			float centerY = get_height() / 2.0f;
+
+			if (get_ellipse_type() == isobus::OutputEllipse::EllipseType::Closed)
+			{
+				if (get_start_angle() == get_end_angle() && get_ellipse_type() != isobus::OutputEllipse::EllipseType::Closed)
+				{
+					/* B.10 / Table B.31 / Ellipse type / NOTE 2:
+           * If type = closed ellipse segment and start and end angle are the same, a
+					 * single line with width = border width shall be drawn from the centre point to the
+					 * point on the border defined by the start and end angles.*/
+					auto angleRadians = degreesToRadians(-((get_start_angle() * 2.0) - 90));
```

Comment:
```suggestion
					auto angleRadians = degreesToRadians(-((get_start_angle() * 2.0f) - 90));
```
Nit, implicit double promotion here. Making the 2.0 into 2.0f fixes it